### DISCUSSION TITLE
Observering "EXTERNAL login refused: rabbit_auth_backend_http failed" issue for 10 mins on setup

### DISCUSSION THREAD

**Discussion by bhakti13111996** [rabbitmq/rabbitmq-server#10484] [Questions] [answered]
Observing below error for 10 mins with rabbitmq 
within these 10 mins observing below error in rabbitmq logs and after retrying multiple times communication is getting established 

![image](https://github.com/rabbitmq/rabbitmq-server/assets/34230934/ea0dec6f-1c72-4f56-a5a6-8a57fa2766b7)

Under what circumstances can these error are expected ?
Any leads would help

**Comment by michaelklishin** [comment]
When the plugin cannot connect to the service it is pointed to. The log message mentions the hostname and the port.

We cannot suggest much with a single log entry. The plugin has [half a dozen of examples](https://github.com/rabbitmq/rabbitmq-server/tree/main/deps/rabbitmq_auth_backend_http/examples) of HTTP-based services. Even then, it's up to you to configure the URLs correctly.

**Reply by bhakti13111996** [reply]
```
Description: ""Server authenticity is not verified since certificate path validation is not enabled""[0m"
572f3cb69309781ce59862b56f04161e68f999ec572e: "2024-01-30T17:51:21.341314140+00:00  [38;5;214m2024-01-30 17:51:21.341044+00:00 [warning] <0.8087.0>      Reason: ""The option {verify, verify_peer} and one of the options 'certfile' or 'certs' are required to enable `this.""[0m"`

```

The setup was operational. After some time, when the server became overloaded with requests, we observed the error mentioned above in the RabbitMQ logs. **We only encountered this error for a duration of 10 minutes. However, afterward, we noticed that the application was running smoothly.**

Furthermore, other services attempting to connect to RabbitMQ are encountering the following error.


`e506e5709d1839dcb342b08bf2c9b0d1d539: "2024-01-30T17:44:23.069409670+00:00  Jan 30 17:44:23  policy[3] MainThread flexsnap.connectors.rabbitmq: INFO - Channel <Channel number=1 CLOSED conn=<SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncSSLTransport object at 0x75dc4d520> params=<ConnectionParameters host=flexsnap-rabbitmq port=5671 virtual_host=/ ssl=True>>> closed unexpectedly:(403, ""ACCESS_REFUSED - access to exchange 'amq.default' in vhost '/' refused for user 'policy.0', backend rabbit_auth_backend_http returned an error: {failed_connect,\n                                                                                           ..."") RMQ channel shutdown"
`

**Reply by michaelklishin** [reply, marked as answer]
There is nothing new in this message, the plugin very clearly logs that it fails to connect to the HTTP endpoints it was configured to use.

I don't know what service does

> Server authenticity is not verified since certificate path validation is not enabled

come from but looks like it may be [Erlang/OTP](https://github.com/erlang/otp/pull/6319)'s TLS implementation. And all it is is a warning.

Focus on investigating what are the HTTP endpoints configured to be used by the `rabbitmq_auth_backend_http` plugin, and why it may not be reachable. We have a [methodology to recommend](https://rabbitmq.com/troubleshooting-networking.html) for troubleshooting networking (it applies well beyond RabbitMQ nodes).

We cannot tell you why your internal HTTP service is not reachable. If you cannot figure it out either, consider [paying for support](https://forms.clickup.com/43297829/f/199b15-647834/SVPNW4U9G69FQCS65P) or hiring someone with more experience in technical operations.
